appinventor/blocklyeditor/src/blocks/helpers.js

```diff
@@ -134,6 +134,12 @@ Blockly.Blocks['helpers_screen_names'] = {
     this.setOutput(true, utils.YailTypeToBlocklyType('text', utils.OUTPUT));
     this.appendDummyInput()
         .appendField(dropdown, 'SCREEN');
+
+    this.workspace.addChangeListener(function(e) {
```

Comment:
Yeah this PR has gotten very messy over time... If I were to rewrite it I'd make it much easier to review haha.

So I had a [listener inside the FieldInvalidDropdown](https://github.com/BeksOmega/appinventor-sources/pull/11/files#diff-a61ef7a00c00fadab2852fa9827eebd8L44) but I realized it wasn't necessary for the helpers_dropdown blocks, so I moved the listeners into the screens & assets fields.

> You should also remove the event listener when the block is disposed, otherwise we will just get a pile of these event listeners for every block added/deleted.
Thanks for the catch! Actually these should be assigned to the onchange property. Those get removed automatically.